homeassistant/components/plugwise/binary_sensor.py

```diff
@@ -89,10 +125,47 @@ def _async_process_data(self):
 
         self._is_on = data[self._binary_sensor]
 
-        self._state = STATE_ON if self._is_on else STATE_OFF
         if self._binary_sensor == "dhw_state":
             self._icon = FLOW_ON_ICON if self._is_on else FLOW_OFF_ICON
         if self._binary_sensor == "slave_boiler_state":
             self._icon = FLAME_ICON if self._is_on else IDLE_ICON
 
         self.async_write_ha_state()
+
+
+class PwNotifySensor(SmileBinarySensor, BinarySensorEntity):
+    """Representation of a Plugwise Notification binary_sensor."""
+
+    def __init__(self, api, coordinator, name, dev_id, binary_sensor):
+        """Set up the Plugwise API."""
+        super().__init__(api, coordinator, name, dev_id, binary_sensor)
+
+        self._attributes = {}
+        for severity in SEVERITIES:
+            self._attributes[f"{severity}_msg"] = None
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
+    @callback
+    def _async_process_data(self):
+        """Update the entity."""
+        notify = self._api.notifications
+
+        self._is_on = False
+        self._icon = NO_NOTIFICATION_ICON
+
+        if notify:
+            self._is_on = True
+            self._icon = NOTIFICATION_ICON
+
+            for dummy, details in notify.items():
+                for msg_type, msg in details.items():
+                    if msg_type not in SEVERITIES:
+                        msg_type = "other"
+
+                    self._attributes[f"{msg_type.lower()}_msg"] = msg
```

Comment:
is it possible for there to be more than one of a type at a time? this logic seems like it would overwrite and just show last msg of each type.

Reply:
I don't think it gives more than one (and/or of each) - @bouwew can you check?

Reply:
@ctalkington discussed offline and we've decided to ensure it wouldn't override, fb8a18f incorporates this